Source/WebCore/platform/graphics/ImageUtilities.h

```diff
@@ -44,5 +44,10 @@ WEBCORE_EXPORT Vector<String> findImagesForTranscoding(const Vector<String>& pat
 // happens while transcoding, a null string will be added to the returned list.
 WEBCORE_EXPORT Vector<String> transcodeImages(const Vector<String>& paths, const String& destinationUTI, const String& destinationExtension);
 
+#if PLATFORM(COCOA)
+WEBCORE_EXPORT bool canDataBeDecodedAsSVG(std::span<const uint8_t>);
+WEBCORE_EXPORT std::optional<Vector<FloatSize>> availableSizesFromData(std::span<const uint8_t>, bool isSVGData);
```

Comment:
I do not understand this interface fully. I think you want to get

1. Whether the encoded data is a bitmap image or SVG image
2. Some metadata about the bitmap image. Here you want to get the size(s) of image frame(s).

I think the first API should say this explicitly like `isSVGImage` or `isSVGImageRep`.
The second API is a little problematic. I think it should be used solely for bitmap image. It does not make sense to send to it the flag `isSVGData` and if it is true it returns an empty `Vector<FloatSize>`. We can do this without calling the function:
```
auto sizes = isSVGData ? Vector<FloatSize>() : availableSizesForBitmapImage(data);
```

Reply:
Updated patch as suggested.